org.civicrm.reqangular/test/decorators/uib-tooltip.decorator.spec.js

```diff
@@ -0,0 +1,230 @@
+              expect(getTooltipVisibility()).toBe(false);
+            });
+          });
+
+          describe('when the opened tooltip is hovered', function () {
+            beforeEach(function () {
+              $triggerElement.trigger('mouseleave');
+              flushTimeout();
+              $tooltipElement.trigger('mouseenter');
+              flushTimeout();
+            });
+
+            it('keeps the tooltip shown', function () {
+              expect(getTooltipVisibility()).toBe(true);
+            });
+
+            describe('when the opened tooltip is unhovered', function () {
+              beforeEach(function () {
+                $tooltipElement.trigger('mouseleave');
+                flushTimeout(2);
+              });
+
+              it('keeps the tooltip shown', function () {
+                expect(getTooltipVisibility()).toBe(false);
+              });
+            });
+          });
+        });
+      });
+
+      describe('touch device', function () {
+        var $overlay;
+
+        beforeEach(function () {
+          simulateTouchDevice();
+
+          $compile($triggerElement)($rootScope.$new());
+          $document.find('body').append($triggerElement);
+
+          $overlay = $triggerElement.find('.tooltip-overlay');
+          $rootScope.$digest();
+        });
+
+        it('creates an properly styled overlay over the trigger element', function () {
+          expect($overlay.length).toBe(1);
+          expect($overlay[0].style.position).toBe('absolute');
+          expect($overlay[0].style.height).toBe('100%');
+          expect($overlay[0].style.height).toBe('100%');
+          expect($overlay[0].style.width).toBe('100%');
+          expect($overlay[0].style.top).toBe('0px');
+          expect($overlay[0].style.left).toBe('0px');
+          expect($overlay[0].style['z-index']).toBe('1');
+        });
+
+        describe('when the overlay is tapped', function () {
+          beforeEach(function () {
+            $overlay.trigger('touchend');
+            flushTimeout();
+
+            $tooltipElement = $document.find('.tooltip-clickable-template');
+          });
+
+          it('shows the tooltip', function () {
+            expect(getTooltipVisibility()).toBe(true);
+          });
+
+          it('hides the overlay', function () {
+            expect(getOverlayVisibility()).toBe(false);
+          });
+
+          describe('when the trigger element is tapped', function () {
+            beforeEach(function () {
+              $triggerElement.trigger('touchend');
+              $rootScope.$digest();
+              flushTimeout(2);
+            });
+
+            it('hides the tooltip', function () {
+              expect(getTooltipVisibility()).toBe(false);
+            });
+
+            it('shows the overlay', function () {
+              expect(getOverlayVisibility()).toBe(true);
+            });
+          });
+
+          describe('when the tooltip clicked', function () {
```

Comment:
`when the tooltip **is** clicked` (or tapped ?)

Reply:
Done.